packages/engine-core/src/common/errors/PrismaClientError.ts

```diff
@@ -0,0 +1,17 @@
+export type PrismaClientErrorInfo = { code: string; cause?: Error }
+
+export abstract class PrismaClientError extends Error {
```

Comment:
I'm introducing this new kind of abstract error so that we can have a clear error standard.